source/slang/slang-lower-to-ir.cpp

```diff
@@ -494,6 +495,13 @@ struct SharedIRGenContext
 
     Dictionary<IntVal*, IRInst*> mapSpecConstValToIRInst;
 
+    // External (imported) unsafeForceInline functions that need to
+    // prelink after lowering.
+    // Key is the mangled name of the function, and value is the
+    // IRInst in the imported module that needs to be cloned into the
+    // current module.
+    Dictionary<String, IRInst*> externalSymbolsToPrelink;
```

Comment:
This obviously works well as-is (no change being requested), but the structure here is different from what I intuitively expected. I would have assumed we would be building up a list of the `IRInst*`s in the *current* module, and then the linking step would do the lookup to find the right instruction in the imported module.

It might be possible to change this so that it actually holds a tuple with both the `IRInst*` in the current module and the `IRInst*` in the imported module, given that the function that adds to this dictionary is also where the `IRInst*` in the current module gets created.

Reply:
Is there a reason this needs to be a dictionary rather than a list? I'm not seeing a point where we do any lookups into it (as opposed to iterating over all entries). If this changes to just a flat list then the mangled name string is probably redundant, since the instruction is expected to have a linkage decoration with a matching string.

Reply:
changed to just a list. We store a list of original inst, because that way we don't have to decide which external inst to pull in during prelinking.